src/services/lineHeaderService.ts

```diff
@@ -46,18 +57,54 @@ class LineHeaderService {
         return LineHeaderFactory.mapExternalLineHeader(queryResult.data.lineHeader);
     };
 
-    public static updateLineHeader = async (lineHeader: ILineHeader) => {
-        await ApiClient.updateObject(endpoints.LINE_HEADER, lineHeader);
-    };
+    public static massEditLineHeaders = async (
+        massEditLineHeaders: IMassEditLineHeader[],
+        oldLineHeaders: ILineHeader[],
+        lineId: string
+    ) => {
+        const added: ILineHeader[] = [];
+        const edited: ILineHeader[] = [];
+        const removed: ILineHeader[] = [];
+        const originals: ILineHeader[] = [];
+        massEditLineHeaders!.forEach((massEditLineHeader: IMassEditLineHeader) => {
+            const currentLineHeader = massEditLineHeader.lineHeader;
+            if (massEditLineHeader.isRemoved) {
+                removed.push(currentLineHeader);
+            } else {
+                const originalLineHeader = currentLineHeader.originalStartDate
+                    ? oldLineHeaders.find(oldLineHeader =>
+                          areDatesEqual(
```

Comment:
I think _.isEqual works here too

Reply:
_.isEqual takes into account that date objects are exactly the same. That "areDatesEqual" compares year, month and date. It doesn't take into account seconds / hours . Even though dates should be converted to "midnightDate" this might not always be the case. 

Maybe in areDatesEqual() method there should be comparison with year / month / date + _.isEqual and if those results are different, it should throw something to prevent errors... additional check would slow down code execution though